mesa/EvoAgent.py

```diff
@@ -0,0 +1,85 @@
+from mesa.agent import *
+import copy
+import numpy as np
+
+class EvoAgent(Agent):
+    """
+    An agent suited for playing in an evolutionary environment.
+    EvoAgent extends Agent to deal with energy, age, death, reproduction and genome
+
+    Attributes:
+    dna: contains the genetic code of the individual, an array of float numbers whose length is determined on creation by the parameter 'genome_len'
+    energy: the energy of the agent
+    age: the age, it is automatically incremented at every step
+    repr_p: the individual probability of reproduction, relevant when using a probabilistic selection model. The individual probability overrides the global probability.
+    die_p: the individual probability of death, relevant when using a probabilistic selection model. The individual probability overrides the global probability.
+    mutation_rate: how much the dna can mutate during reproduction
+    """
+    def __init__(self, unique_id,model,energy=False,genome_len=10,repr_p=False,die_p=False,mutation_rate=1.0):
+        """
+        Args:
+        unique_id: the unique identifier of this agent
+        energy: the starting energy. If False, the energy is ignored. Defaults to False.
+        genome_len: the size of the dna
+        repr_p: the individual probability of reproduction
+        die_p: the individual probability of death
+        """
+        super().__init__(unique_id,model)
+        self.dna = np.random.random(genome_len)
+        self.energy=energy
+        self.age=0
+        self.repr_p=repr_p
+        self.die_p=die_p
+        self.mutation_rate=mutation_rate
+        self.deceased=False
+
+    def step(self):
+        """
+        Performs one step of simulation
+
+        Side effect: age is increased by 1
+        """
+        self.age+=1
+
+    def duplicate(self):
```

Comment:
Or rather, clone in the strictest sense is DNA->DNA copy, but duplicate is for molecule->molecule copy. The former is just the recipe, which doesn't even include the epigenome, the latter's definition includes all that is necessary.